Python_Scripts/compareStateTrajectories_BioModels_1.py

```diff
@@ -114,44 +119,63 @@ def compStaTraj_BioModels():
                         solver.setStabilityLimitFlag(False)
 
                     # Simulate model
+                    print("Simulating with AMICI")
                     sim_data = amici.runAmiciSimulation(model, solver)
 
+                    # If AMICI failed, we cannot write any trajectory
+                    if sim_data['status'] < 0:
+                        continue
+
                     # np.set_printoptions(threshold=8, edgeitems=2)
-                    for key, value in sim_data.items():
-                        print('%12s: ' % key, value)
+                    #for key, value in sim_data.items():
+                    #    print('%12s: ' % key, value)
 
                     # Get state trajectory
                     state_trajectory = sim_data['x']
 
-                    # Delete all trajectories for boundary conditions
-                    delete_counter = 0
-                    all_properties = sbml_model.getModel()
-                    for iSpec in range(0, all_properties.getNumSpecies()):
-                        all_species = all_properties.getSpecies(iSpec)
-                        if all_species.getBoundaryCondition() == True:
-                            state_trajectory = state_trajectory.transpose()
-                            if delete_counter == 0:
-                                state_trajectory = np.delete(state_trajectory, iSpec, 0)
-                            else:
-                                state_trajectory = np.delete(state_trajectory, iSpec - delete_counter, 0)
-                            state_trajectory = state_trajectory.transpose()
-                            delete_counter = delete_counter + 1
+                    # prune out constant species for trajectory comparison
+                    # open sbml file
+                    sbml_model = libsbml.readSBML(sbml_path)
+                    delete_ind = [
+                        iSpec for iSpec in
+                        range(sbml_model.getModel().getNumSpecies())
+                        if sbml_model.getModel().getSpecies(
+                            iSpec).getBoundaryCondition()
+                           is True]
+                    state_trajectory = np.delete(
+                        state_trajectory, delete_ind, 1)
 
                     # Adjustment for the 'Froehlich2018' model
+                    # TODO Y Why?
                     if iModel == 'Froehlich2018':
-                        state_trajectory = np.delete(state_trajectory, np.s_[1177:1178], axis=1)
+                        state_trajectory = np.delete(
+                            state_trajectory, np.s_[1177:1178], axis=1)
```

Comment:
This is pruning out one species... Why written like this and why done at all?

Reply:
yes ... feel free to investigate ;)